### ISSUE TITLE
WKT with reference units in km breaks gdalwarp reprojection in 3.6.3

### ISSUE THREAD

**Issue by glaroc** [OSGeo/gdal#8601]
Running gdalwarp on the this file  [maps_occ.zip](https://github.com/OSGeo/gdal/files/13078612/maps_occ.zip), which has a custom WKT CRS with units in kilometers

````
gdalwarp -t_srs EPSG:32198 -of COG -co COMPRESS=DEFLATE maps_occ.tif out.tif
````

On 3.6.3, this outputs a GeoTIFF with improper extents, appearing to improperly interpret units in kilometers as meters. On 3.4.1, the output is fine.

**Comment by jratike80** [comment]
The WKT is:

```
PROJCRS["unknown",
    BASEGEOGCRS["NAD83",
        DATUM["North American Datum 1983",
            ELLIPSOID["GRS 1980",6378137,298.257222101004,
                LENGTHUNIT["metre",1]]],
        PRIMEM["Greenwich",0,
            ANGLEUNIT["degree",0.0174532925199433]],
        ID["EPSG",4269]],
    CONVERSION["Lambert Conic Conformal (2SP)",
        METHOD["Lambert Conic Conformal (2SP)",
            ID["EPSG",9802]],
        PARAMETER["Latitude of false origin",47,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8821]],
        PARAMETER["Longitude of false origin",-75,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8822]],
        PARAMETER["Latitude of 1st standard parallel",49,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8823]],
        PARAMETER["Latitude of 2nd standard parallel",62,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8824]],
        PARAMETER["Easting at false origin",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8826]],
        PARAMETER["Northing at false origin",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8827]]],
    CS[Cartesian,2],
        AXIS["easting",east,
            ORDER[1],
            LENGTHUNIT["kilometre",1000,
                ID["EPSG",9036]]],
        AXIS["northing",north,
            ORDER[2],
            LENGTHUNIT["kilometre",1000,
                ID["EPSG",9036]]]]
```

**Comment by jratike80** [comment]
Please add the proper and improper extents.

**Comment by jratike80** [comment]
GDAL v. 3.6.3 is not supported so please test with 3.7 or 3.8dev.

**Comment by glaroc** [comment]
We were pulling images from https://hub.docker.com/r/osgeo/gdal, which had 3.6.3 as the latest version. I now realize that the images are hosted on Github for newer versions. I confirm that this seems to have been fixed in 3.7 and the bounds of the converted file are good.
